### PR TITLE
Add a new doc about TSO

### PR DESCRIPTION
**Pull request by hfxsd**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.5 (TiDB 7.5 versions)
- [x] v7.4 (TiDB 7.4 versions)
- [x] v7.3 (TiDB 7.3 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs/pull/14630
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tso.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+The `>> 18` operation signifies a bitwise [right shift](/functions-and-operators/bit-functions-and-operators.md) by 18 bits, which is used to filter out the physical timestamp. Because the physical timestamp is expressed in milliseconds, deviating from the more customary UNIX timestamp format measured in seconds, you need to divide it by 1000 to convert it into a format compatible with [`FROM_UNIXTIME()`](/functions-and-operators/date-and-time-functions.md). Essentially, this process aligns with the functionality of `TIDB_PARSE_TSO()`.
+
+`>> 18` 操作表示按位[右移](/functions-and-operators/bit-functions-and-operators.md) 18 位，用于滤除物理时间戳。由于物理时间戳是以毫秒为单位的，与更常用的以秒为单位的 UNIX 时间戳格式不同，因此需要除以 1000 才能将其转换为与 [`FROM_UNIXTIME()`](/functions-and-operators/date-and-f-time-functions.md) 兼容的格式。这个过程基本与 `TIDB_PARSE_TSO()` 的功能一致。
+
+你还可以将二进制中的逻辑时间戳 `000000000000000100`（即十进制中的 `4`）过滤掉。
+
+你也可以通过 CLI 工具查看时间戳，命令如下：
+
+```shell
+$ tiup ctl:v7.4.0 pd tso 443852055297916932
```
Comment (hfxsd) [inline tso.md:92]:
```suggestion
$ tiup ctl:v7.1.0 pd tso 443852055297916932
```

File: tso.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+SELECT @ts;
++--------------------+
+| @ts                |
++--------------------+
+| 443852055297916932 |
++--------------------+
+1 row in set (0.00 sec)
+```
+
+注意由于 TSO 时间戳是分配给事务的，所以该时间戳是在带有 (`BEGIN; ...; ROLLBACK`) 的事务中生成的。
```
Comment (hfxsd) [inline tso.md:27]:
```suggestion
注意由于 TSO 时间戳是分配给事务的，所以该时间戳是在带有 `BEGIN; ...; ROLLBACK` 的事务中生成的。
```

File: tso.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+SELECT TIDB_PARSE_TSO_LOGICAL(443852055297916932);
++--------------------------------------------+
+| TIDB_PARSE_TSO_LOGICAL(443852055297916932) |
++--------------------------------------------+
+|                                          4 |
++--------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+下面示例展示了 TSO 时间戳的细节：
```
Comment (hfxsd) [inline tso.md:52]:
```suggestion
下面示例展示了 TSO 时间戳的二进制细节：
```

File: tso.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                            @ts: 443852055297916932
+         UNIX_TIMESTAMP(NOW(6)): 1693161835.502954
+               (@ts >> 18)/1000: 1693161221.6870
+FROM_UNIXTIME((@ts >> 18)/1000): 2023-08-27 20:33:41.6870
+                         NOW(6): 2023-08-27 20:43:55.502954
+                  @ts & 0x3FFFF: 4
+1 row in set (0.00 sec)
+```
+
+`>> 18` 操作表示按位[右移](/functions-and-operators/bit-functions-and-operators.md) 18 位，用于滤除物理时间戳。由于物理时间戳是以毫秒为单位的，与更常用的以秒为单位的 UNIX 时间戳格式不同，因此需要除以 1000 将其转换为与 [`FROM_UNIXTIME()`](/functions-and-operators/date-and-time-functions.md) 兼容的格式。这个过程基本与 `TIDB_PARSE_TSO()` 的功能一致。
```
Comment (hfxsd) [inline tso.md:79]:
```suggestion
`>> 18` 操作表示按位[右移](/functions-and-operators/bit-functions-and-operators.md) 18 位，用于滤除物理时间戳。由于物理时间戳以毫秒为单位，与更常用的以秒为单位的 UNIX 时间戳格式不同，因此需要除以 1000 将其转换为与 [`FROM_UNIXTIME()`](/functions-and-operators/date-and-time-functions.md) 兼容的格式。这个过程基本与 `TIDB_PARSE_TSO()` 的功能一致。
```

File: tso.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
+---
+title: TiDB 中的 TimeStamp Oracle (TSO)
+summary: 了解 TiDB 中的 TimeStamp Oracle (TSO)。
+---
+
+# TiDB 中的 TimeStamp Oracle (TSO)
+
+在 TiDB 中，Placement Driver (PD) 在为各种集群组件分配时间戳方面发挥着关键作用。这些时间戳用于为事务和数据分配时间标记。该机制对于在 TiDB 中启用 [Percolator](https://research.google.com/pubs/pub36726.html) 模型至关重要。Percolator 模型用于支持多版本并发控制（Multi-Version Concurrency Control, MVCC）和[事务管理](/transaction-overview.md)。
+
+下面示例显示了如何获取 TiDB 中的 TSO：
```
Comment (hfxsd) [inline tso.md:10]:
```suggestion
下面示例显示了如何获取 TiDB 中的当前的 TSO：
```

File: TOC.md [hunk 6]
```diff
@@ -631,6 +631,7 @@
 - 参考指南
   - 架构
     - [概述](/tidb-architecture.md)
+    - [TSO](/tso.md)
     - [存储](/tidb-storage.md)
     - [计算](/tidb-computing.md)
     - [调度](/tidb-scheduling.md)
```
Comment (hfxsd) [inline TOC.md:637]:
```suggestion
    - [存储](/tidb-storage.md)
    - [计算](/tidb-computing.md)
    - [调度](/tidb-scheduling.md)
    - [TSO](/tso.md)
```

**Review by tonyxuqqi — APPROVED** [review]

File: tso.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
+---
+title: TiDB 中的 TimeStamp Oracle (TSO)
+summary: 了解 TiDB 中的 TimeStamp Oracle (TSO)。
+---
+
+# TiDB 中的 TimeStamp Oracle (TSO)
+
+在 TiDB 中，Placement Driver (PD) 在为各种集群组件分配时间戳方面发挥着关键作用。这些时间戳用于为事务和数据分配时间标记。该机制对于在 TiDB 中启用 [Percolator](https://research.google.com/pubs/pub36726.html) 模型至关重要。Percolator 模型用于支持多版本并发控制（Multi-Version Concurrency Control, MVCC）和[事务管理](/transaction-overview.md)。
```
Comment (qiancai) [inline tso.md:8]:
```suggestion
在 TiDB 中，Placement Driver (PD) 承担着 TSO 时间戳分配器的角色，负责为集群内各组件分配时间戳。这些时间戳用于为事务和数据分配时间标记。该分配机制对于在 TiDB 中启用 [Percolator](https://research.google.com/pubs/pub36726.html) 模型至关重要。Percolator 模型用于支持多版本并发控制（Multi-Version Concurrency Control, MVCC）和[事务管理](/transaction-overview.md)。
```

File: tso.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+SELECT @ts;
++--------------------+
+| @ts                |
++--------------------+
+| 443852055297916932 |
++--------------------+
+1 row in set (0.00 sec)
+```
+
+注意由于 TSO 时间戳依据事务进行分配的，所以该时间戳是在带有 `BEGIN; ...; ROLLBACK` 的事务中生成的。
```
Comment (qiancai) [inline tso.md:27]:
```suggestion
注意由于 TSO 时间戳是按事务分配的，所以需要从包含 `BEGIN; ...; ROLLBACK` 的事务中获取时间戳。
```

File: tso.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+                            @ts: 443852055297916932
+         UNIX_TIMESTAMP(NOW(6)): 1693161835.502954
+               (@ts >> 18)/1000: 1693161221.6870
+FROM_UNIXTIME((@ts >> 18)/1000): 2023-08-27 20:33:41.6870
+                         NOW(6): 2023-08-27 20:43:55.502954
+                  @ts & 0x3FFFF: 4
+1 row in set (0.00 sec)
+```
+
+`>> 18` 操作表示按位[右移](/functions-and-operators/bit-functions-and-operators.md) 18 位，用于提取物理时间戳。由于物理时间戳以毫秒为单位，与更常见的以秒为单位的 UNIX 时间戳格式不同，因此需要除以 1000 将其转换为与 [`FROM_UNIXTIME()`](/functions-and-operators/date-and-time-functions.md) 兼容的格式。这个过程与 `TIDB_PARSE_TSO()` 的功能一致。
```
Comment (qiancai) [inline tso.md:81]:
```suggestion
`>> 18` 操作表示按位[右移](/functions-and-operators/bit-functions-and-operators.md) 18 位，用于提取物理时间戳。由于物理时间戳以毫秒为单位，与更常见的以秒为单位的 UNIX 时间戳格式不同，因此需要除以 1000 将其转换为与 [`FROM_UNIXTIME()`](/functions-and-operators/date-and-time-functions.md) 兼容的格式。这个转换过程与 `TIDB_PARSE_TSO()` 的功能一致。
```

File: tso.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+0000011000101000111000010001011110111000110111                    ← 前 46 位是物理时间戳
+                                              000000000000000100  ← 后 18 位是逻辑时间戳
+```
+
+TSO 时间戳由两部分组成：
+
+- 物理时间戳：自 1970 年 1 月 1 日以来的 UNIX 时间戳，单位为毫秒。
+- 逻辑时间戳：递增计数器，用于需要在一毫秒内使用多个时间戳的情况，或某些事件可能触发时钟进程逆转的情况。在这些情况下，物理时间戳会保持不变，而逻辑时间戳保持递增。该机制可以确保 TSO 时间戳的完整性，确保时间戳始终递增而不会倒退。
+
+你可以在 SQL 中更深入地查看 TSO 时间戳，示例如下：
```
Comment (qiancai) [inline tso.md:67]:
```suggestion
你可以通过 SQL 语句更深入地查看 TSO 时间戳，示例如下：
```

File: tso.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+```shell
+$ tiup ctl:v7.1.0 pd tso 443852055297916932
+```
+
+```
+system:  2023-08-27 20:33:41.687 +0200 CEST
+logic:   4
+```
+
+可以看到以 `system:` 开头的行中的物理时间戳，以 `logic:` 开头的行中的逻辑时间戳。
```
Comment (qiancai) [inline tso.md:96]:
```suggestion
可以看到，物理时间戳在以 `system:` 开头的行中，逻辑时间戳在以 `logic:` 开头的行中。
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**